src/lib.rs

```diff
@@ -108,27 +106,48 @@ pub enum FoundCrate {
 /// The returned crate name is sanitized in such a way that it is a valid rust identifier. Thus,
 /// it is ready to be used in `extern crate` as identifier.
 pub fn crate_name(orig_name: &str) -> Result<FoundCrate, Error> {
-    let manifest_dir =
-        PathBuf::from(env::var("CARGO_MANIFEST_DIR").map_err(|_| Error::CargoManifestDirNotSet)?);
+    struct Cache {
+        manifest_dir: String,
+        manifest_path: PathBuf,
+        manifest: Table,
+    }
 
-    let cargo_toml_path = manifest_dir.join("Cargo.toml");
+    static CACHE: OnceCell<Cache> = OnceCell::new();
+    let cache = CACHE.get_or_try_init(|| {
+        let manifest_dir =
+            env::var("CARGO_MANIFEST_DIR").map_err(|_| Error::CargoManifestDirNotSet)?;
 
-    if !cargo_toml_path.exists() {
-        return Err(Error::NotFound(manifest_dir.into()));
-    }
+        let manifest_path = Path::new(&manifest_dir).join("Cargo.toml");
+
+        if !manifest_path.exists() {
+            return Err(Error::NotFound(manifest_dir.into()));
+        }
+
+        let manifest = open_cargo_toml(&manifest_path)?;
+
+        Ok(Cache {
+            manifest_dir,
+            manifest_path,
+            manifest,
+        })
+    })?;
 
-    let cargo_toml = open_cargo_toml(&cargo_toml_path)?;
+    let manifest_dir = env::var("CARGO_MANIFEST_DIR").map_err(|_| Error::CargoManifestDirNotSet)?;
```

Comment:
Can be moved above initializing the cache to reuse it there.

Reply:
It's already done inside the cache initialization, it's just being re-read here as an extra sanity check, to ensure it's the same between all invocations of `crate_name` within one process.

Reply:
Yeah I know. However, when you initialize the cache you could just reuse this manifest dir.

Reply:
Oh, I see what you mean. Sure.